bucket-metadata.go

```diff
@@ -0,0 +1,83 @@
+//
+// MinIO Object Storage (c) 2022 MinIO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package madmin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// ExportBucketMetadata makes an admin call to export bucket metadata of a bucket
+func (adm *AdminClient) ExportBucketMetadata(ctx context.Context, bucket string) (io.ReadCloser, error) {
+	path := fmt.Sprintf(adminAPIPrefix + "/export-bucket-metadata")
+	queryValues := url.Values{}
+	queryValues.Set("bucket", bucket)
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodGet, requestData{
+			relPath:     path,
+			queryValues: queryValues,
+		},
+	)
+	if err != nil {
+		closeResponse(resp)
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	if resp.Body == nil {
+		return nil, errors.New("body is nil")
+	}
+
+	return resp.Body, nil
+}
+
+// ImportBucketMetadata makes an admin call to set bucket metadata of a bucket from imported content
+func (adm *AdminClient) ImportBucketMetadata(ctx context.Context, bucket string, content []byte) error {
+	path := fmt.Sprintf(adminAPIPrefix + "/import-bucket-metadata")
+	queryValues := url.Values{}
+	queryValues.Set("bucket", bucket)
+
+	resp, err := adm.executeMethod(ctx,
+		http.MethodPut, requestData{
+			relPath:     path,
+			queryValues: queryValues,
+			content:     content,
+		},
+	)
+	if err != nil {
+		closeResponse(resp)
```

Comment:
You can just do `defer closeResponse(resp)`  you are leaking this in the Good case.